cmd/executor/cmd/root.go

```diff
@@ -175,6 +175,9 @@ func addKanikoOptionsFlags() {
 	RootCmd.PersistentFlags().VarP(&opts.Labels, "label", "", "Set metadata for an image. Set it repeatedly for multiple labels.")
 	RootCmd.PersistentFlags().BoolVarP(&opts.SkipUnusedStages, "skip-unused-stages", "", false, "Build only used stages if defined to true. Otherwise it builds by default all stages, even the unnecessaries ones until it reaches the target stage / end of Dockerfile")
 	RootCmd.PersistentFlags().BoolVarP(&opts.RunV2, "use-new-run", "", false, "Experimental run command to detect file system changes. This new run command does no rely on snapshotting to detect changes.")
+	RootCmd.PersistentFlags().StringVarP(&opts.Git.Branch, "git-branch", "", "", "Branch to clone if build context is a git repository")
```

Comment:
Thanks @caarlos0 for the PR. 
i was wondering if we could instead of adding this flags, try a config file or one flag 
- points to git config.
- a flag which accepts key-value pair.

I just feel the number of flags to kaniko are increasing a lot.

WDYT?

Reply:
I agree, and that sounds good indeed.

Will work on it.

Reply:
Hacked something real quick, lmk if that looks like what you had in mind...